integrations_test.go

```diff
@@ -224,3 +225,15 @@ func TestContextifyFramesNonexistingFilesShouldNotDropFrames(t *testing.T) {
 	contextifiedFrames := cfi.contextify(frames)
 	assertEqual(t, len(contextifiedFrames), len(frames))
 }
+
+func TestContextifyFramesConcurrentCacheUpdates(t *testing.T) {
+	cfi := contextifyFramesIntegration{
```

Comment:
Does this test repro the issue before the fix is applied?

Reply:
Yes. I made sure it fails first. Although I wasnt able to do this for reads, thus I didnt change them anywhere.